### PR TITLE
Add some docs for the SQL injection detection

### PR DESCRIPTION
**Pull request by CaspianA1**
Going to merge this once [this](https://github.com/dbos-inc/eslint-plugin/pull/12) PR is merged

### CODE REVIEW

File: docs/api-reference/static-analysis.md [hunk 1]
```diff
@@ -31,19 +31,16 @@ DBOS uses several techniques to ensure that static analysis is as productive as
...
 
-### Installing and configuring `eslint`
-
-::::tip
-If you got started with the [quickstart](../getting-started/quickstart.md), `eslint` and required plugins are already installed.
-Plugins to support TypeScript and detect common vulnerabilities are automatically installed with `@dbos-inc/eslint-plugin` as dependencies and do not need to be installed separately.
-::::
+### Installing and configuring the plugin
 
 To install the `eslint` package and the DBOS plugin:
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:36]:
```suggestion
To install the DBOS eslint plugin:
```

File: docs/api-reference/static-analysis.md [hunk 2]
```diff
@@ -31,19 +31,16 @@ DBOS uses several techniques to ensure that static analysis is as productive as
 
 Many DBOS-suggested coding practices can be enforced by a combination of `eslint` plugins and rule configurations.
 
-### Installing and configuring `eslint`
-
-::::tip
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:36]: We still want to mention that "if you got started with the quickstart, the `@dbos-inc/eslint-plugin` is already installed for you."

Comment (CaspianA1) [reply docs/api-reference/static-analysis.md:36]: Resolved

File: docs/api-reference/static-analysis.md (hunk 1 above)
Comment (CaspianA1) [reply docs/api-reference/static-analysis.md:36]: Done

File: docs/api-reference/static-analysis.md [hunk 3]
```diff
@@ -155,6 +154,44 @@ These plugins are enabled by default:
...
+```ts
+export class Greetings {
+  @Transaction()
+  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
+    await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`);
+  }
+}
+```
+
+- A LR-value is a literal-reducible value, or a value that is known to be literal at some level.
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:186]: Let's remove this paragraph as it's an implementation detail. Replace it with plain language instead.

File: docs/api-reference/static-analysis.md [hunk 4]
```diff
@@ -155,6 +154,44 @@ These plugins are enabled by default:
...
 One custom rule from DBOS, `@dbos-inc/dbos-static-analysis`, is provided in the [`@dbos-inc/eslint-plugin`](https://github.com/dbos-inc/eslint-plugin) package.  This rule is enabled by default.
 
+___
+
+Running a database query with a string that's vulnerable to SQL injection will result in an error.
+This would typically happen inside of a [transaction](https://docs.dbos.dev/tutorials/transaction-tutorial).
+- SQL injection happens when a bad actor puts SQL code as a field into something like an online form,
+and if a programmer builds a raw query from SQL and this data, the bad actor's supposed data may allow them to run
+arbitrary SQL commands over your database.
+- To avoid injection, use prepared statements. But if you accidentally make yourself vulnerable to injection, DBOS is here to save you!
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:164]: Replace prepared statement -> parameterized query

File: docs/api-reference/static-analysis.md (hunk 3 above)
Comment (CaspianA1) [reply docs/api-reference/static-analysis.md:186]: Done

File: docs/api-reference/static-analysis.md (hunk 4 above)
Comment (CaspianA1) [reply docs/api-reference/static-analysis.md:164]: Done

File: docs/api-reference/static-analysis.md [hunk 5]
```diff
@@ -155,6 +154,40 @@ These plugins are enabled by default:
...
+```ts
+export class Greetings {
+  @Transaction()
+  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
+    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
+  }
+}
+```
+
+Here's how you would make SQL injection possible:
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:176]:
```suggestion
This example is vulnerable to SQL injection:
```

File: docs/api-reference/static-analysis.md [hunk 6]
```diff
@@ -155,6 +154,40 @@ These plugins are enabled by default:
...
+```ts
+export class Greetings {
+  @Transaction()
+  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:180]:
```suggestion
  static async VulnerableGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
```

File: docs/api-reference/static-analysis.md [hunk 7]
```diff
@@ -155,6 +154,40 @@ These plugins are enabled by default:
...
+    await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`);
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:181]:
```suggestion
    await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`); // Don't do this!
```

File: docs/api-reference/static-analysis.md [hunk 8]
```diff
@@ -155,6 +154,40 @@ These plugins are enabled by default:
...
+export class Greetings {
+  @Transaction()
+  static async VulnerableGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
+    await ctxt.client.raw(`INSERT INTO greetings (name, note) VALUES (${friend}, ${note})`); // Don't do this!
+  }
+}
+```
+
+- The plugin will raise a potential SQL injection error if your query string is either directly or indirectly built up of a nonliteral component.
+- For example, if your query is a format string that has formatting parameters of things like a function call, a parameter, or something else like that, the plugin will let you know that you're vulnerable to injection.
```
Comment (qianl15) [inline docs/api-reference/static-analysis.md:187]:
```suggestion
- For example, if your query is a format string parameterized by a function call, input variable, or other non-literal inputs, the plugin will let you know that you're vulnerable to injection.
```

**Review by qianl15 — APPROVED** [review]
Nice work!

**Pull request merged**